WcaOnRails/app/models/post.rb

```diff
@@ -26,7 +26,7 @@ def edit_path
     if is_crash_course_post?
       Rails.application.routes.url_helpers.delegate_crash_course_edit_path
     else
-      Rails.application.routes.url_helpers.edit_post_path(slug)
+      Rails.application.routes.url_helpers.edit_post_path(id: slug)
```

Comment:
I'm not really sure, maybe this has something to do with the fact that we call the `url_helpers` from the model, whereas in the others cases it's from the controller or the view, where the locale is set in the `default_url_options`?

Here is a relevant explanation I could find in the referenced issue in rails:

> Ughh, I know what this is - it's the optimised url generation in Rails 4 which does a simple substitution on the path pattern if it meets some simple criteria. Adding an option to default_url_options disables it which is why if fixed your problem.